app/components/ui/button.tsx

```diff
@@ -58,13 +58,18 @@ const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
     const Comp = asChild ? Slot : 'button';
     return (
       <Comp
-        className={cn(buttonVariants({ variant, size, className }))}
+        className={cn('relative', buttonVariants({ variant, size, className }))}
         ref={ref}
         disabled={loading || disabled}
         {...props}
       >
         {loading ? (
-          <LoaderCircle className="h-5 w-5 animate-spin text-muted-foreground" />
+          <>
+            <span className="invisible">{children}</span>
```

Comment:
keeping the button content in dome when loading but just make hidden makes sure that the width of the button stays the same